addons/project/static/src/views/project_task_list/project_task_list_controller.js

```diff
@@ -1,19 +1,19 @@
 /** @odoo-module */
 
 import { ListController } from "@web/views/list/list_controller";
-import { DeleteSubtasksConfirmationDialog } from "@project/components/delete_subtasks_confirmation_dialog/delete_subtasks_confirmation_dialog";
+import { subTaskDeleteConfirmationMessage } from "@project/views/project_task_form/project_task_form_controller";
 
 export class ProjectTaskListController extends ListController {
-    async onDeleteSelectedRecords() {
-        if (!Math.max(...this.model.root.selection.map((record) => record.data.subtask_count ))) {
-            return super.onDeleteSelectedRecords();
+
+    get deleteConfirmationDialogProps() {
+        const deleteConfirmationDialogProps = super.deleteConfirmationDialogProps;
+        const subTaskCount = this.model.root.selection.some(task => task.data.subtask_count > 0)
+        if (!subTaskCount) {
```

Comment:
Very small suggestion:
```suggestion
        const hasSubtasks = this.model.root.selection.some(task => task.data.subtask_count > 0)
        if (!hasSubtasks) {
```